python/src/aac/templates/engine.py

```diff
@@ -125,6 +132,16 @@ def write_generated_templates_to_file(generated_files: list[TemplateOutputFile])
         )
 
 
+def _handle_template_error(error_reason: str, callback: Callable, *args, **kwargs):
+    try:
+        return callback(*args, **kwargs)
+    except TemplateError as template_error:
+        raise AacTemplateError(
+            f"{template_error.__class__.__name__} occurred running {__package__}.{callback.__name__}:\n"
+            f"{error_reason}\n{template_error}"
+        )
```

Comment:
This is great! The template errors were so much more helpful. 
```suggestion
        error_message = f"{error_reason}\n{template_error}"

        if hasattr(template_error, "filename"):
            error_message = f"{error_reason}\n{os.path.basename(template_error.filename)}:L{template_error.lineno} {template_error.message}"

        raise AacTemplateError(
            f"{template_error.__class__.__name__} occurred running {__package__}.{callback.__name__}:\n{error_message}"
        )
```

This suggestion would add some additional info for the user which may be easier to diagnose template errors with.
```
gen_plant_uml: plugin_failure

TemplateSyntaxError occurred running aac.templates._load_templates_from_env:
The error occurred while loading the templates from 'templates/component/' for the 'aac.plugins.gen_plant_uml' package
expected token 'end of statement block', got '-'
```

to
```
gen_plant_uml: plugin_failure

TemplateSyntaxError occurred running aac.templates._load_templates_from_env:
The error occurred while loading the templates from 'templates/component/' for the 'aac.plugins.gen_plant_uml' package
component_diagram.puml.jinja2:L10 expected token 'end of statement block', got '-'
```